src/averageAges.js

```diff
@@ -56,6 +77,38 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const result = [];
+  let sum = 0;
+
+  if (arguments.length > 1) {
+    people.forEach((elem) => {
+      for (const elem2 of people) {
```

Comment:
Instead of using a combination of forEach and for loops, use array methods like .filter and .map